notebooks/Import_species_data.qmd

```diff
@@ -0,0 +1,576 @@
+
+### **In Summary:**
+
+The existence of multiple biodiversity data portals reflects the complexity and diversity of global biodiversity itself. Each portal contributes uniquely by focusing on different regions, species groups, or community engagement strategies. Together, they create a comprehensive network that enhances our understanding of biodiversity, supports conservation efforts, and fosters global collaboration among scientists, policymakers, and the public.
+
+## 1. Set the working directory and make a folder to store data.
+
+```{r set_up}
+# Set Workspace as the current working directory
+workspace <- getwd()
+
+env_data_dir <- file.path(workspace, "env_data")
+
+# Create the 'raw_data' directory if it doesn't exist
+if (!dir.exists(env_data_dir)) {
+  dir.create(env_data_dir, recursive = TRUE)
+  cat("Directory 'env_data' created successfully.\n")
+} else {
+  cat("Directory 'env_data' already exists.\n")
+}
+
+```
+
+## 2. Install and load essential libraries.
+
+```{r load_libraries}
+# List of packages to check, install if needed, and load
+packages <- c("dplyr", "sf", "ggplot2", "rnaturalearth", "rnaturalearthdata")
+
+# Install missing packages and load them
+for (pkg in packages) {
+  if (!pkg %in% installed.packages()[, "Package"]) {
+    install.packages(pkg)
+  }
+  library(pkg, character.only = TRUE)
+}
+
+```
+
+## 3. Open source species data portals introduction
+
+We will introduce three of the most popular and largest biodiversity data portals that provide users with the ability to download their data via their platforms or several R packages. Please note, exploring these datasets on their own platforms is always the most effective method for accessing their data.
+
+**Exemplar species: Splendid Fairy-Wren (*Malurus splendens*)**
+
+<img src="https://raw.githubusercontent.com/EcoCommons-Australia-2024-2026/ec-notebook_site/main/images/ala_sfw.jpeg" alt="ALA Image" width="400"/>
+
+We are going to use a beautiful bird species - Splendid Fairy-Wren (Malurus splendens) as a representative example to compare the data you can get from the three datasets. (Photographer: pimelea, more copyright information [here](https://biocache.ala.org.au/occurrences/a87e9029-e829-4e10-9988-1e74614af81b))
+
+## 3.1 GBIF - the Global Biodiversity Information Facility
+
+GBIF—the Global Biodiversity Information Facility—is an international network and data infrastructure funded by the world's governments. It aims to provide anyone, anywhere, open access to data about all types of life on Earth.
+
+**You need to register with GBIF to get "username", "password", and "registered email address" to access their dataset via their R package "rgbif".**
+
+```{r load_libraries_3}
+# Install the "rgbif" R package
+
+install.packages("rgbif") # CRAN version
+
+library(rgbif)
```

Comment:
This whole section should be done at the top when you are setting the environment